list-agreements.lua

```diff
@@ -66,26 +96,49 @@ function is_denied(agr)
     return denied
 end
 
-function generate_output_guild(agr)
-    if is_satisfied(agr) == true then
-        print("Establish a "..get_location_name(agr).." for the "..get_guildhall_profession(agr).." guild, as agreed on "..get_petition_date(agr).." (satisfied)")
-    elseif is_denied(agr) == true then
-        print("Establish a "..get_location_name(agr).." for the "..get_guildhall_profession(agr).." guild, as agreed on "..get_petition_date(agr).." (denied)")
-    else
-        print("Establish a "..get_location_name(agr).." for the "..get_guildhall_profession(agr).." guild, as agreed on "..get_petition_date(agr))
+function is_expired(agr)
+    local expired = false
+    agr_age = get_petition_age(agr)
+    if agr_age[1] ~= 0 then
+        expired = true
     end
+    return expired
 end
 
-function generate_output_temple(agr)
-    if is_satisfied(agr) == true then
-        print("Establish a "..get_location_name(agr).." for "..get_religion_name(agr)..", as agreed on "..get_petition_date(agr).." (satisfied)")
-    elseif is_denied(agr) == true then
-        print("Establish a "..get_location_name(agr).." for "..get_religion_name(agr)..", as agreed on "..get_petition_date(agr).." (denied)")
+--universal handler
+function generate_output(agr,loctype)
+    local loc_name = get_location_name(agr.details[0].data.Location.tier,loctype)
+    local agr_age = get_petition_age(agr)
+    output_str = "Establish a ".. loc_name.." for \""..get_agr_party_name(agr)
```

Comment:
It's generally cleaner in Lua to use single quotes for string literals. Then you don't have to escape the double quotes for quoted strings in the output. e.g. `'Establish a '..loc_name..'for "'..get_agr_party_name(agr)`